flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/extraction/FunctionSignatureTemplate.java

```diff
@@ -44,19 +45,24 @@ final class FunctionSignatureTemplate {
 
     final @Nullable String[] argumentNames;
 
+    final @Nullable Boolean[] argumentOptionals;
```

Comment:
Currently, since optional is not included as part of inputStrategy, argumentName and argumentOptional are not involved in the calculation of equal and hashcode. Therefore, if optional participates in the calculation of hashCode and equals here, it may result in duplicate inputStrategies.

Reply:
I think it's not related to InputStrategy here. `FunctionSignatureTemplate` just a POJO object that describes the method. The current implementation is counterintuitive and contradicts the way users utilize it.